react-components/stories/Image360CollectionContainer.stories.tsx

```diff
@@ -16,17 +22,47 @@ const meta = {
 export default meta;
 type Story = StoryObj<typeof meta>;
 
-const sdk = createSdkByUrlToken();
-
 export const Main: Story = {
   args: {
-    collectionId: { siteId: 'Hibernia_RS2' }
+    collectionId: { siteId: 'c_RC_2', transform: new Matrix4() }
   },
-  render: ({ collectionId }) => (
-    <RevealContext sdk={sdk} color={new Color(0x4a4a4a)}>
-      <RevealCanvas>
-        <Image360CollectionContainer collectionId={collectionId} />
-      </RevealCanvas>
-    </RevealContext>
+  render: ({ collectionId, styling }) => (
+    <RevealStoryContainer color={new Color(0x4a4a4a)}>
+      <Image360CollectionContainerStoryContent
+        collectionId={collectionId}
+        styling={styling}
+        transform={collectionId.transform}
+      />
+    </RevealStoryContainer>
   )
 };
+
+type CadModelContainerStoryContentProps = {
+  collectionId: AddImageCollection360Options;
+  transform?: Matrix4;
+  styling?: ImageCollectionModelStyling;
+};
+
+const Image360CollectionContainerStoryContent = ({
+  collectionId,
+  transform,
+  styling
+}: CadModelContainerStoryContentProps): ReactElement => {
+  const modelsLoadedRef = useRef(0);
+  const cameraNavigationActions = useCameraNavigation();
+  const onLoad = (): void => {
+    modelsLoadedRef.current++;
+    if (modelsLoadedRef.current === 1) {
```

Comment:
Hmmm interesting. So looks like the onLoad is triggered twice or I read it wrong?
That is why it has this counter?

Maybe the add360Collection (inside the useEffect) and therefore the getOrAdd360Collection are being called twice or doesnt make no sense what I'm saying? :D :D

Reply:
I can remove the condition here. In this case it will only be called once. If you have multiple models, then we need to make sure all are loaded before fitting the camera, which is the reason why the check is here. For one model though it doesn't really make sense. I'll remove it.